ast/ksp/src/main/kotlin/me/tatarka/kotlin/ast/KSAst.kt

```diff
@@ -530,6 +531,8 @@ private class KSAstType private constructor(
         // we check for error first because KotlinPoet will throw an exception
         return if (type.isError) {
             typeRef.toString()
+        } else if (type.declaration is KSTypeParameter) {
+            typeRef.toString()
```

Comment:
This won't completely fix this as a type param can be arbitrarily nested, for example changing your test from
```
val c: C
```
to
```
val c: () -> Unit
```
will cause the same exception.

Reply:
maybe we could wrap `toTypeName()` in a try/catch and fallback to `typeRef.toString()` for now?

As an aside, I really wish ksp's `toString()` properly showed type params, as I wish we wouldn't need to rely on kotlinpoet at all. I submitted a pr to fix this but it's been ignored 😐 https://github.com/google/ksp/pull/1737

Reply:
I updated the test to reflect that and changed to try catch.